FORMS.md

```diff
@@ -0,0 +1,192 @@
+
+### Handling Server Errors
+
+Server errors related to form submission should appear in the Form Alert above the submit button. They should not appear in growls or other kinds of alerts. Additionally, as best practice moving forward server errors should never solely do the work that frontend validation can also do. This means that any error that can be validated in the frontend should be validated in the frontend and backend.
+
+Note: This is not meant to suggest that we should avoid validating in the backend if the frontend already validates.
+
+Note: There are edge cases where some server errors will inevitably relate to specific fields in a form with other fields unrelated to that error. We had trouble coming to a consensus on exactly how this edge case should be handled (e.g. show inline error, clear on blur, etc). For now, we will show the server error in the form alert and not inline (so the “fix the errors” link will not be present). In those cases, we will still attempt to inform the user which field needs attention, but not highlight the input or display an error below the input. We will be on the lookout for our first validation in the server that could benefit from being tied to a specific field and try to come up with a unified solution for all errors.
+
+## Form Submission
+### Submit Button Disabling
+
+Submit buttons shall not be disabled or blocked from being pressed in most cases. We will allow the user to submit a form and point them in the right direction if anything needs their attention.
+
+The only time we won’t allow a user to press the submit button is when we have submitted the form and are waiting for a response (e.g. from the API). In this case we will show a loading indicator and additional taps on the submit button will have no effect. This is handled by the Form component and will also ensure that a form cannot be submitted multiple times.
+
+## Using Form.js
+
+The example below shows how to use [Form.js](https://github.com/Expensify/App/blob/c5a84e5b4c0b8536eed2214298a565e5237a27ca/src/components/Form.js) in our app. You can also refer to [Form.stories.js](https://github.com/Expensify/App/blob/c5a84e5b4c0b8536eed2214298a565e5237a27ca/src/stories/Form.stories.js) for more examples.
+
+```
+validate: (values) => {
+    const errors = {};
+    if (!values.routingNumber) {
+        errors.routingNumber = 'Please enter a routing number';
+    }
+    if (!values.accountNumber) {
+        errors.accountNumber = 'Please enter an account number';
+    }
+    return errors;
+}
+
+onSubmit: (values) => {
+    setTimeout(() => {
+        alert(`Form submitted!`);
+        FormActions.setIsSubmitting('TestForm', false);
+    }, 1000);
+}
+
+<Form
+    formID="testForm"
+    submitButtonText="Submit"
+    validate={this.validate}
+    onSubmit={this.onSubmit}
+>
+    <View>
```

Comment:
Why is this in a View and not the TextInput below

Reply:
Added a description. Essentially we are just showing that form inputs can be nested in other components.